src/unix/pipe.c

```diff
@@ -134,6 +134,9 @@ void uv__pipe_close(uv_pipe_t* handle) {
 int uv_pipe_open(uv_pipe_t* handle, uv_file fd) {
   int err;
 
+  if (uv__fd_exists(handle->loop, fd))
+    return UV_EEXIST;
```

Comment:
Are runtime errors what we want or should they be asserts?  It's ultimately a programmer bug because said programmer is trying to do something that libuv doesn't support - but if it's an assert, we can't test it easily. :-)

Reply:
I think Node.js would prefer if it errored rather than asserted so we can choose what action to take. :D

Reply:
I think a runtime error would be better.